pmd-core/src/main/java/net/sourceforge/pmd/AbstractPropertySource.java

```diff
@@ -127,25 +118,22 @@ public boolean hasDescriptor(PropertyDescriptor<?> descriptor) {
         return result;
     }
 
-    /**
-     * {@inheritDoc}
-     */
+
     @Override
     public <T> void setProperty(PropertyDescriptor<T> propertyDescriptor, T value) {
```

Comment:
That's a nice idea! I'll implement the first version, I think it's a bit clearer with the more generic interface